src/client.js

```diff
@@ -540,32 +540,36 @@ governing permissions and limitations under the License.
     const _getPublicUrl = async (fileRes) => {
       return await client.NLWS.xtkFileRes.create(fileRes).getURL()
     }
+
     return {
       /**
         * This is the exposed/public method for fileUploader instance which will do all the processing related to the upload process internally and returns the promise containing all the required data.
         * @param file
         * @returns {Promise<{name: string, md5: string, type: string, size: string, url: string}>}
         */
       upload: async (file) => {
-        return new Promise(async (resolve, reject) => {
+        return new Promise((resolve, reject) => {
           if (!Util.isBrowser()) {
             reject('File uploading is only supported in browser based calls.');
           }
-          try {
-            var data = new FormData()
-            data.append('file_noMd5', file)
-            //TODO: Needs to be refactored after cookie issue get resolved.
-            const response = await fetch(`${client._connectionParameters._endpoint}/nl/jsp/uploadFile.jsp`, {
-              processData: false,
-              credentials: 'include',
-              method: 'POST',
-              body: data,
-              headers: {
-                'x-security-token': client._securityToken,
-                'Cookie': '__sessiontoken=' + client._sessionToken,
-              }
-            })
-            const okay = await response.text()
+          //try {
+          var data = new FormData();
+          data.append('file_noMd5', file);
+          //TODO: Needs to be refactored after cookie issue get resolved.
+
+          const httpPromise = client._makeHttpCall({
```

Comment:
This won't send cookies to the server, this is why I added a TODO comment to refactor when the uploadFile.jsp will do the authentication based on HTTP headers but not cookies.

Reply:
Yes, I know the point here is 1/ to use _transport instead of fetch and 2/ to fix the "double async" issue that I mention in my email. Consider this change as needed to pass the test. You can implement it as you see fit in the main branch